enclave/asym_keys.c

```diff
@@ -480,188 +480,110 @@ void oe_free_key(
     }
 }
 
-static oe_result_t _copy_to_from_host(
-    bool to_host,
-    const void* data,
-    size_t data_size,
-    uint8_t** out)
+uint32_t oe_get_public_key_by_policy_ecall(
+    uint32_t seal_policy,
+    const oe_asymmetric_key_params_t* key_params,
+    void* key_buffer,
+    size_t key_buffer_size,
+    size_t* key_buffer_size_out,
+    void* key_info,
+    size_t key_info_size,
+    size_t* key_info_size_out)
 {
     oe_result_t result = OE_UNEXPECTED;
-    uint8_t* out_local = NULL;
-
-    if (data == NULL)
+    struct
     {
-        *out = NULL;
-        result = OE_OK;
-        goto done;
-    }
+        uint8_t* key_buffer;
+        size_t key_buffer_size;
+        uint8_t* key_info;
+        size_t key_info_size;
+    } arg;
```

Comment:
Consider initializing the arg struct here. Otherwise if line 509 raises OE_INVALID_PARAMETER, the value of arg is junk.

Reply:
Line 509 does not examine the args structure. The args structure is cleared immediately afterwards with **memset**.

Reply:
What I meant was that if line 509 raises OE_INVALID_PARAMETER, then control will jump to done label and we would execute `oe_free(arg.key_buffer)` on an uninitialized arg.

Reply:
Thanks for the explanation. I researched "{ 0 }" initializers and am switching to those. Thanks.